### PR TITLE
Jbruno/reqestor stats

### PR DESCRIPTION
**Pull request by JeanetteBruno**
adding scheduler reporting job/task stats by requestor.  see:
http://go/vizurl/768945  (number of running jobs by requestor)
http://go/vizurl/768947 (number of running tasks by requestor)
http://go/vizurl/768949 (number of waiting tasks by requestor)
http://go/vizurl/768951 (number of waiting jobs by requestor)
(make sure the timespan covers 11/7 19:00 - 21:00 UTC)

### CODE REVIEW

File: sched/scheduler/stateful_scheduler.go [hunk 1]
```diff
@@ -406,25 +424,100 @@ func (s *statefulScheduler) step() {
 	s.killJobs()
 	s.scheduleTasks()
 
-	remaining := 0
-	waitingToStart := 0
+	s.updateStats()
+}
+
+/*
+ * update the stats monitoring values:
```
Comment (dgdonut) [inline sched/scheduler/stateful_scheduler.go]: Nit that we are mixing use of "//", "/* ... */", and "/* * * * * */" styles

File: sched/scheduler/stateful_scheduler.go [hunk 2]
```diff
@@ -406,25 +424,100 @@ func (s *statefulScheduler) step() {
...
+		}
+
+		remainingTasks += (len(job.Tasks) - job.TasksCompleted)
 		if job.TasksCompleted+job.TasksRunning == 0 {
-			waitingToStart += 1
+			jobsWaitingToStart += 1
+			s.requestorsCounts[requestor]["jobsWaitingToStart"]++
+			s.requestorsCounts[requestor]["numWaitingTasks"] += len(job.Tasks)
+		} else if job.getJobStatus() == sched.InProgress {
+			s.requestorsCounts[requestor]["jobsRuning"]++
```
Comment (dgdonut) [inline sched/scheduler/stateful_scheduler.go]: "jobsRuning" -> "jobsRunning" (at least it's consistent!)

File: sched/scheduler/stateful_scheduler.go [hunk 3]
```diff
@@ -406,25 +424,100 @@ func (s *statefulScheduler) step() {
...
+			s.requestorsCounts[requestor]["numRunningTasks"] += job.TasksRunning
+			s.requestorsCounts[requestor]["numWaitingTasks"] += len(job.Tasks) - job.TasksCompleted - job.TasksRunning
+		}
+
+	}
+
+	// publish the requestor stats
+	for requestor, counts := range s.requestorsCounts {
+		s.stat.Gauge(fmt.Sprintf("%s_%s", stats.SchedNumRunningJobsGauge, requestor)).Update(int64(
```
Comment (dgdonut) [inline sched/scheduler/stateful_scheduler.go]: Assuming this format "<req>_<num>" is easy enough to group on in viz?

**Review by dgdonut — APPROVED** [review]

File: sched/scheduler/stateful_scheduler.go (hunk 1 above)
Comment (JeanetteBruno) [reply sched/scheduler/stateful_scheduler.go]: fixed

File: sched/scheduler/stateful_scheduler.go (hunk 2 above)
Comment (JeanetteBruno) [reply sched/scheduler/stateful_scheduler.go]: fixed

File: sched/scheduler/stateful_scheduler.go [hunk 4]
```diff
@@ -427,28 +426,30 @@ func (s *statefulScheduler) step() {
...
+//for each unique requestor count:
+//. number of tasks running
+//. number of tasks waiting
+//. number of jobs running or waiting to start
 func (s *statefulScheduler) updateStats() {
 	remainingTasks := 0
 	jobsWaitingToStart := 0
 
 	// reset current counts to 0
+	jobsRunnningKey := "jobsRunning"
```
Comment (ryancouto) [inline sched/scheduler/stateful_scheduler.go]: extra n

Comment (JeanetteBruno) [reply sched/scheduler/stateful_scheduler.go]: fixed

**Pull request merged**